tests/test_main.py

```diff
@@ -103,3 +104,17 @@ def mocked_main(*args):
 
     with pytest.raises(FakeFailureError):
         xonsh.main.main()
+
+
+def test_xonsh_failback_script_from_file(shell, monkeypatch):
```

Comment:
too much monkeypatching going on, feels like we are testing more that we should on these tests.
failback only makes sense on interactive mode anyway.